kernel_tuner/observers/observer.py

```diff
@@ -57,4 +57,15 @@ def process_output(self, answer, output):
         """
         pass
 
+class PrologueObserver(BenchmarkObserver):
+    """Observer that measures something in a seperate kernel invocation prior to the normal benchmark."""
 
+    @abstractmethod
+    def before_start(self):
```

Comment:
The regular `BenchmarkObserver` also has a method called `before_start`, but that one is called before each iteration that the kernel is run (not just in the prologue). Maybe we should rename this one to `before_start_prologue`?

Reply:
It is good to make a distinction between these two, indeed. However, since `before_start` is an abstract method of the `PrologueObserver`, the `_prologue` suffix may also be considered a bit redundant. I don't have a strong preference here.

Reply:
I would agree with @csbnw in this case. The type PrologueObserver already signals that there is specialization from the more generic BenchmarkObserver.

Reply:
Let's stick to `before_start`.

Reply:
But this means that `before_start` and `after_finish` are also called during the regular benchmark iterations right? Not only in the prologue? The `PrologueObserver`s will also be in the list of regular observers in `core.py`.

Reply:
I think you are right. Simply removing the observer from the `dev.observers` isn't a solution, because of how the metrics are evaluated. This needs a bit more investigation.

Reply:
You are right, renaming to `prologue_start` and `prologue_finish` solves this issue. I personally like these names a bit better than `before_start_prologue` and `after_finish_prologue`, but don't mind renaming it if you prefer.

Reply:
I am resolving this conversation, as it is no longer relevant since Ben restructured things.